webui/react/src/pages/InteractiveTask.tsx

```diff
@@ -21,47 +22,95 @@ enum PageView {
   TASK_LOGS = 'Task Logs'
 }
 
+const DEFAULT_PAGE_TITLE = 'Tasks - Determined';
+
+const getTitleState = (commandState: CommandState | undefined): string => {
+  if (!commandState){
+    return DEFAULT_PAGE_TITLE;
+  }
+  const commandStateTitleMap = {
+    [CommandState.Pending]: 'Pending',
+    [CommandState.Assigned]: 'Assigned',
+    [CommandState.Pulling]: 'Pulling',
+    [CommandState.Running]: DEFAULT_PAGE_TITLE,
+    [CommandState.Terminating]: 'Terminating',
+    [CommandState.Terminated]: 'Terminated',
+    [CommandState.Starting]: 'Starting',
+  };
+  const title = commandStateTitleMap[commandState];
+  if(commandState !== CommandState.Terminated && commandState !== CommandState.Running){
+    return title + '...';
+  }
+  return title;
+
+};
+
 export const InteractiveTask: React.FC = () => {
 
   const [ pageView, setPageView ] = useState<PageView>(PageView.IFRAME);
   const { taskId, taskName, taskResourcePool, taskUrl, taskType } = useParams<Params>();
-
+  const [ taskState, setTaskState ] = useState<CommandState>();
   const storeDispatch = useStoreDispatch();
+  const { ui } = useStore();
+
+  const handleMessage = (event: MessageEvent) => {
+    const messageFromSameOrigin = window.location.origin === event.origin;
+    if(event?.data?.commandState && messageFromSameOrigin){
```

Comment:
**super nit:** `if (` space between if it's easy to change. I keep forget to add that lint rule